snap/snapenv/snapenv.go

```diff
@@ -63,6 +67,7 @@ func GetBasicSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
 		"SNAP={{.SnapPath}}",
 		"SNAP_DATA=/var{{.SnapPath}}",
+		"SNAP_SHARED_DATA={{CleanPath (printf \"/var%s/../shared\" .SnapPath)}}",
```

Comment:
Yeah I wasn't a huge fan either, but the only nice alternative I could think of was to stop being clever using `SnapPath` for everything and actually specify each of these explicitly, which is quite a bit more invasive (e.g. create an `info.BasePath()` to get `/snap/snapname` without the version, and so on). Would that be better?